pkg/resource/deploy/source_eval.go

```diff
@@ -453,6 +454,10 @@ func (d *defaultProviders) handleRequest(req providers.ProviderRequest) (provide
 	return ref, nil
 }
 
+func (d *defaultProviders) RegisterDefaultProvider(provider providers.Reference) {
+	d.providers[provider.String()] = provider
```

Comment:
Is provider.String() right here? if yes why don't we reuse that logic in setProviderAsDefault?

Reply:
Nah, this is broken I think, and currently unused which is why I didn't noticed in testing.  I'll just remove this.